packages/react/src/primitives/shared/styleUtils.ts

```diff
@@ -126,37 +125,25 @@ export const convertStylePropsToStyleObj: ConvertStylePropsToStyleObj = ({
   breakpoint,
   breakpoints,
 }) => {
-  ComponentPropsToStylePropsMapKeys.filter(
-    (stylePropKey) => !isNullOrEmptyString(props[stylePropKey])
-  ).forEach((stylePropKey: any) => {
-    let value = props[stylePropKey];
-    // if styleProp is a DesignToken use its toString()
-    if (isDesignToken(value)) {
-      value = value.toString();
-    } else {
-      value = getValueAtCurrentBreakpoint(value, breakpoint, breakpoints);
-    }
-    const reactStyleProp = ComponentPropsToStylePropsMap[stylePropKey];
-    style = { ...style, [reactStyleProp]: value };
-  });
-  return style;
-};
-
-/**
- * Filter out known style props to prevent errors adding invalid HTML attributes
- * @param props
- * @returns non styled props
- */
-export const useNonStyleProps = (props: ViewProps) => {
-  return React.useMemo(() => {
-    const nonStyleProps = {};
-    Object.keys(props).forEach((propKey) => {
-      if (!(propKey in ComponentPropsToStylePropsMap)) {
-        nonStyleProps[propKey] = props[propKey];
+  const nonStyleProps = {};
+  Object.keys(props)
+    .filter((propKey) => props[propKey] !== null)
+    .forEach((stylePropKey) => {
```

Comment:
This isn't necessarily a style prop key, so should keep generic propKey variable name.
```suggestion
    .forEach((propKey) => {
```